From b846f12c0276bd67e97ae7ede67e3246f7e376ac Mon Sep 17 00:00:00 2001
From: Jorge Ruesga <jorge@ruesga.com>
Date: Fri, 10 Apr 2015 22:26:01 +0200
Subject: [PATCH 04/20] unified-email: respect swipe user setting

Don't start swipe if the user has disabled it by settings

Change-Id: I79b2d694219dc6497fbe85d7608bd36ee48a42d7
Signed-off-by: Jorge Ruesga <jorge@ruesga.com>
---
 src/com/android/mail/ui/SwipeableListView.java | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/com/android/mail/ui/SwipeableListView.java b/src/com/android/mail/ui/SwipeableListView.java
index 39429d03b..6adad0475 100644
--- a/src/com/android/mail/ui/SwipeableListView.java
+++ b/src/com/android/mail/ui/SwipeableListView.java
@@ -170,14 +170,15 @@ public class SwipeableListView extends ListView implements Callback, OnScrollLis
         if (mScrolling) {
             return super.onInterceptTouchEvent(ev);
         } else {
-            return (!mPreventSwipesEntirely && mSwipeHelper.onInterceptTouchEvent(ev))
-                    || super.onInterceptTouchEvent(ev);
+            return (mEnableSwipe && !mPreventSwipesEntirely
+                    && mSwipeHelper.onInterceptTouchEvent(ev)) || super.onInterceptTouchEvent(ev);
         }
     }
 
     @Override
     public boolean onTouchEvent(MotionEvent ev) {
-        return (!mPreventSwipesEntirely && mSwipeHelper.onTouchEvent(ev)) || super.onTouchEvent(ev);
+        return (mEnableSwipe && !mPreventSwipesEntirely
+                && mSwipeHelper.onTouchEvent(ev)) || super.onTouchEvent(ev);
     }
 
     @Override
-- 
2.17.1

